pkg/provisioner/worker.go

```diff
@@ -0,0 +1,86 @@
+package provisioner
+
+import (
+	"fmt"
+
+	"github.com/iver-wharf/wharf-cmd/pkg/worker"
+	v1 "k8s.io/api/core/v1"
+)
+
+// Worker contains the data of a worker, an abstraction over k8s pods which is
+// rougly comparable to a group of Docker containers sharing
+// namespaces/volumes.
+type Worker struct {
+	ID     string
+	Name   string
+	Status worker.Status
+}
+
+// WorkerList contains a slice of workers and a count field for the size of the
+// array.
+type WorkerList struct {
+	Items []Worker
+	Count int
+}
+
+func convertPodListToWorkerList(podList *v1.PodList) WorkerList {
+	if podList == nil {
+		return WorkerList{}
+	}
+	return WorkerList{
+		Items: convertPodsToWorkers(podList.Items),
+		Count: len(podList.Items),
+	}
+}
```

Comment:
Excess type. Strongly suggest to remove these and rely on the caller to just call the `convertPodsToWorkers` on the `podList.Items`.

Important to note that a nil slice acts as an empty slice, so this abstraction is not needed in Go code.

```go
len([]string(nil)) // => 0
```